components/raftstore/src/store/peer.rs

```diff
@@ -1694,16 +1698,22 @@ where
                 _ => {}
             }
             self.on_leader_changed(ss.leader_id, self.term());
-            // TODO: it may possible that only the `leader_id` change and the role
-            // didn't change
-            ctx.coprocessor_host
-                .on_role_change(self.region(), ss.raft_state);
+            ctx.coprocessor_host.on_role_change(
+                self.region(),
+                RoleChange {
+                    state: ss.raft_state,
+                    leader_id: ss.leader_id,
+                    prev_lead_transferee: self.lead_transferee,
```

Comment:
I don't get it. In what case when `lead_transferee` has value while it's not equal to `vote`?

Reply:
`lead_transferee` could be stale. For example, If the leadership has transfered from A to B, and then starts to transfer to C quickly.

Reply:
In this case, `vote` should still be the correct value.

Reply:
Yes, however `vote` isn't the tranferee. We hope to only handle the leadership transferring case.

Reply:
What's the point of transferee in this case? If what you want is the next peer to visit, then `vote` should be used instead of `lead_transferee` as no matter who the leader tries to transfer leader to, the next leader is likely the one it votes for.